src/ModalForm.js

```diff
@@ -0,0 +1,109 @@
+import React, {PropTypes} from 'react';
+import {
+  Button,
+  Modal,
+  FormGroup,
+  ControlLabel,
+  FormControl,
+  HelpBlock
+} from 'react-bootstrap';
+
+function FieldGroup({ id, label, help, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+      {help && <HelpBlock>{help}</HelpBlock>}
+    </FormGroup>
+  );
+}
+
+const ModalForm = React.createClass({
+
+  getInitialState() {
+    return {
+      showModal: false,
+    };
+  },
+
+  close() {
+    this.setState({ showModal: false });
+  },
+
+  open() {
+    this.setState({ showModal: true });
+  },
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const {title, description, buttonText, url} = this.state;
```

Comment:
these values are undefined here, so they're being passed as undefined values to the onSubmit function